pkg/cloud-init/cloud-init.go

```diff
@@ -162,52 +163,20 @@ func ReadCloudInitVolumeDataSource(vmi *v1.VirtualMachineInstance, secretSourceD
 	return nil, nil
 }
 
-// resolveNoCloudSecrets is looking for CloudInitNoCloud volumes with UserDataSecretRef
-// requests. It reads the `userdata` secret the corresponds to the given CloudInitNoCloud
-// volume and sets the UserData field on that volume.
-//
-// Note: when using this function, make sure that your code can access the secret volumes.
-func resolveNoCloudSecrets(vmi *v1.VirtualMachineInstance, secretSourceDir string) error {
-	volume := findCloudInitNoCloudSecretVolume(vmi.Spec.Volumes)
-	if volume == nil {
-		return nil
-	}
-
-	baseDir := filepath.Join(secretSourceDir, volume.Name)
-	var userDataError, networkDataError error
-	var userData, networkData string
-	if volume.CloudInitNoCloud.UserDataSecretRef != nil {
-		userData, userDataError = readFirstFoundFileFromDir(baseDir, []string{"userdata", "userData"})
-	}
-	if volume.CloudInitNoCloud.NetworkDataSecretRef != nil {
-		networkData, networkDataError = readFirstFoundFileFromDir(baseDir, []string{"networkdata", "networkData"})
-	}
-	if userDataError != nil && networkDataError != nil {
-		return fmt.Errorf("no cloud-init data-source found at volume: %s", volume.Name)
-	}
-
-	if userData != "" {
-		volume.CloudInitNoCloud.UserData = userData
-	}
-	if networkData != "" {
-		volume.CloudInitNoCloud.NetworkData = networkData
-	}
+func isNotNoCloudAccessCredential(accessCred v1.AccessCredential) bool {
+	return accessCred.SSHPublicKey != nil && accessCred.SSHPublicKey.PropagationMethod.NoCloud == nil
+}
 
-	return nil
+func isNotCloudDriveAccessCredential(accessCred v1.AccessCredential) bool {
```

Comment:
nit: This should be called `isNotConfigDriveAccessCredential`

Reply:
Same, change to logic and name to `isConfigDriveAccessCredential`?

Reply:
2 against 1, reverted it back
/cc @lyarwood

ref https://github.com/kubevirt/kubevirt/pull/10231#discussion_r1303986821

Reply:
I think we are making the same point here, the name of the function and logic within still don't match and are confusing when called through `isAccessCredentialValidFunc` later. We also typically test for the positive case within `is` functions or methods. I still think this needs to be written like the following:

```go
func isNoCloudAccessCredential(accessCred v1.AccessCredential) bool {
	return accessCred.SSHPublicKey != nil && accessCred.SSHPublicKey.PropagationMethod.NoCloud != nil
}
```

That then allows you to have the following later that actually makes sense:

```go
if !isAccessCredentialValidFunc(accessCred) { 
    continue
}
```